packages/neuron-wallet/src/controllers/wallets.ts

```diff
@@ -77,6 +77,13 @@ class WalletsController {
     changeAddressNumber: number
   }): ChannelResponse<WalletData> => {
     try {
+      const id = WalletsController.service.getAll().findIndex(wallet => wallet.name === name)
```

Comment:
Check this in data level is better, data level should ensure wallet name not duplicated.

Reply:
I will move to wallet services.Store level is atom.

Reply:
If service throw an error, does controller should catch it and return status and msg ? I'm not sure, @Keith-CY please check it, 3Q

Reply:
It will be caught in walletController.create